gpsgen.py

```diff
@@ -1,9 +1,516 @@
+        "MAD": pyproj.Proj(init=CITIES["MAD"]["proj_map"])
+}
+
+worker_sequence = count(start=1, step=1)
+
+task_sequence = count(start=1, step=1)
+
+city_switcher = cycle(CITIES.keys())
+
+
+def next_worker_id():
+    return next(worker_sequence)
+
+def next_task_id():
+    return next(task_sequence)
+
+def get_current_day():
+    today = datetime.date.today()
+    return "{}{:02d}{:02d}".format(today.year,today.month,today.day)
+
+
+
+"""
+Not sure about this could be interesting, using it as
+Task = namedtuple("Task", ['task_id'])
+>>> task = Task(next_task_id())
+**Note currently not using this, if we dont need no sync to a real database,
+lets keep this, unused.
+"""
+
+class Task():
+    def __init__(self):
+        self.task_id = next_task_id()
+
+"""
+I changed the state machine to iterate over this namedtuples, in which I
+will store 1/100 to change odds to change next state. Quick dirty solution.
+I also stored the code and a name for a posible representation in a redis key
+"""
+Free = namedtuple("Free", ['odds', 'code', 'name'])
+ToProvider = namedtuple("ToProvider", ['odds', 'code', 'name'])
+ToCustomer = namedtuple("ToCustomer", ['odds', 'code', 'name'])
+"""
+## Instantiation for the class variable, lets start thinking the odds will be
+inmutable, thats why I stored them in a namedtuple,  want to keep this data
+structure tight, quick and dirty again.
+"""
+free = Free(6,0,'free') # this means 6/100 chances to change to --> ToProvider state
+to_provider = ToProvider(2,1,'to_provider') # 2/100 chances to change to --> ToCustomer state
+to_customer = ToCustomer(2,2,'to_customer') # 2/100 chances to change to --> Free state
+
+
+class Worker():
+    states = [free, to_provider, to_customer]
+
+    def __init__(self, worker_id, city, coord, task_id, state='free'):
+        """
+        In this proof of concept every Worker has a ACTION linked even if it
+        is <Free>, as well as the a last coord, gps coordinates for displaying
+        the last valid position for the worker.
+        Speed is a vector np array it allow us to multiply matrixes
+        """
+        self.worker_id = worker_id
+        self.coord = coord
+        self.city = city
+        self.state_machine = cycle(self.states)
+        self.current_state = state
+        self.task_id = task_id
+        self.speed = np.array((0,0))
+        self.map_convert = MapProj[self.city]
+        self.trigger_state_generator()
+
+
+    def _get_id(self):
+        return self.worker_id
+
+    def trigger_state_generator(self):
+        """
+        Just semantic alias to start the generator(cycle)
+        """
+        self.current_state = next(self.state_machine)
+        return
+
+    def switch_state(self):
+        self.current_state = next(self.state_machine)
+        return
+
+    def get_position(self):
+        return self.coord
+
+    def __str__(self):
+        return "{city} -> {worker_id} {task_id} {coord}".format(**vars(self))
```

Comment:
i did not get this rigth, istead of:
`def __str__(self):
+        return "{.city} -> {.worker_id} {.task_id} {.coord}".format(self)`
i get a IndexError: tuple index out of range

`    def __str__(self):
        s = "{city} -> {worker_id} {task_id} {coord}"
        return s.format(**self.__dict__)
`
i think i am missing something here